File: meal_plan_api/models/deliveryModel.js
const mongoose = require('mongoose');
const Schema = mongoose.Schema;

const DeliveryItemSchema = new Schema({
  meal: { type: Schema.Types.ObjectId, ref: 'Meal', required: true },
  status: { 
    type: String, 
    enum: ['scheduled', 'skipped', 'moved'], 
    default: 'scheduled' 
  },
});

const DeliverySchema = new Schema({
  userId: { type: Schema.Types.ObjectId, ref: 'User', required: true },
  deliveryDate: { type: Date, required: true },
  timeSlot: { type: String, required: true },
  address: { type: String, required: true }, // The selected address nickname
  deliveryType: { 
    type: String, 
    enum: ['delivery', 'pickup'], 
    default: 'delivery' 
  },
  items: [DeliveryItemSchema],
});

module.exports = mongoose.model('Delivery', DeliverySchema);

File: meal_plan_app/lib/models/delivery_model.dart
import 'meal_model.dart';

class Delivery {
  final String id;
  final DateTime deliveryDate;
  String timeSlot;
  String address;
  String deliveryType;
  final List<DeliveryItem> items;

  Delivery({
    required this.id,
    required this.deliveryDate,
    required this.timeSlot,
    required this.address,
    required this.deliveryType,
    required this.items,
  });

  factory Delivery.fromJson(Map<String, dynamic> json) {
    return Delivery(
      id: json['_id'],
      deliveryDate: DateTime.parse(json['deliveryDate']),
      timeSlot: json['timeSlot'],
      address: json['address'],
      deliveryType: json['deliveryType'],
      items: List<DeliveryItem>.from(
        json['items'].map((x) => DeliveryItem.fromJson(x)),
      ),
    );
  }
}

class DeliveryItem {
  final String id;
  final Meal meal;
  String status;

  DeliveryItem({required this.id, required this.meal, required this.status});

  factory DeliveryItem.fromJson(Map<String, dynamic> json) {
    return DeliveryItem(
      id: json['_id'],
      meal: Meal.fromJson(json['meal']),
      status: json['status'],
    );
  }
}